### DISCUSSION TITLE
redundant % in the pet search output on Cygwin

### DISCUSSION THREAD

**Discussion by AnabasisXu** [knqyf263/pet#327] [Q&A]
Here is a sample setup:

```toml
  2 [[Snippets]]
  3   Description = "hello"
  4   Filename = "D:\\HOME\\dotfiles\\pet\\snippet.toml"
  5   Output = ""
  6   Tag = []
  7   command = "echo hello"

```

Then I search for this snippet, and there is a % at the end of the output.

```bash
❯ pet search
echo hello%     
```

Maybe something wrong with my Cygwin? I tested on Termux and there is no such redundant %.

**Comment by RamiAwar** [comment]
Hmm not sure about this one. Could you do me a favor and see what this outputs on Cygwin? My hypothesis here is that for some reason this newline is not being added on Cygwin, and so the shell itself feels the need to add a % marking the end of the output.

https://github.com/knqyf263/pet/blob/e356762320793655ae51c9a3276f295c265c5c02/cmd/search.go#L34

If you're curious about the new line, more info here: https://unix.stackexchange.com/a/302666

**Comment by AnabasisXu** [comment]
>Could you do me a favor and see what this outputs on Cygwin? 

Sorry I am relatively new to programming, which command should I run on Cygwin? 

I later found that this is probably because of `stty -ixon`, which is from the `README`:

```
cat .zshrc
function pet-select() {
  BUFFER=$(pet search --query "$LBUFFER")
  CURSOR=$#BUFFER
  zle redisplay
}
zle -N pet-select
stty -ixon
bindkey '^s' pet-select
```

I tried resetting with `stty sane` but did not work. Then I restarted Cygwin/Zsh, % was gone. 

For a while it is well and good after I removed the above `stty -ixon` from my `.zshrc`. Unfortunately, when trying to confirm whether it is `stty` that caused the trouble, some how I cannot get rid of % by restarting Cygwin. 

This is what I got with 'stty -a'

```
❯ stty -a
speed 38400 baud; rows 24; columns 80; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>;
eol2 = <undef>; swtch = ^Z; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R;
werase = ^W; lnext = ^V; discard = ^O; min = 1; time = 0;
-parenb -parodd -cmspar cs8 -hupcl -cstopb cread -clocal -crtscts
-ignbrk brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl ixon -ixoff
-iuclc ixany imaxbel iutf8
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0
isig icanon iexten echo echoe echok -echonl -noflsh -tostop echoctl echoke
-flusho
```
